### ISSUE TITLE
Vote not picked up by Query Node - User unable to release stake

### ISSUE THREAD

**Issue by mochet** [Joystream/joystream#3727]
## Problem
* A user wants to release their stake from voting in an election. The problem is the vote does not show in the `Past Votes` screen: https://dao.joystream.org/#/election/past-votes
* Despite using different queries it looks like the Query Node never picked up this vote or it somehow disappeared

## Details
* The user voted using the address `5HWKe5kQKPuKopnWkfwMdULGNsZimaxaLv7HuHjfy2yir3f1`
* This address is the root account for a membership (ID: 458) and this address is also used as the rewards destination for the same member who works for the Operations Beta WG
* The vote shows in `chain state`:
```
referendum.votes: CastVoteOf
{
  commitment: 0x40b4bc679e61c9d1183775d09195d4387cd4831153a8dfa97e59c5629f279b30
  cycle_id: 6
  stake: 500,000
  vote_for: 515
}
```

* Using the query `query {voteCastEvents{inBlock,castVote{id,commitment,electionRoundId,stake,stakeLocked,castBy,voteForId}}}` no results are found for this vote
* Using the query `query {voteRevealedEvents{inBlock,id,castVoteId,castVote{electionRoundId,castBy,stake,votePower,stakeLocked,commitment,voteForId,voteFor{id,memberId,votePower}}}}` no results are found for this vote
* The user showed a screenshot in which the locked stake is assigned for `Voting` on the `My Accounts` screen, so it does show on this screen: 
![image](https://user-images.githubusercontent.com/50830161/167466060-12eca154-6aa5-4ed2-815e-350f191afa57.png)

**Comment by traumschule** [comment]
in which block was the vote revealed?

**Comment by mochet** [comment]
> in which block was the vote revealed?

Don't have it (unless there is some easy way to get it without the QN that I'm not aware of)

**Comment by mnaamani** [comment]
Regarding the query node, perhaps query used reached limit and so the record you were looking didn't show up.
A more targeted query shows the expected event:
```
query {
  voteCastEvents(where: {castVote: {castBy_eq: "5HWKe5kQKPuKopnWkfwMdULGNsZimaxaLv7HuHjfy2yir3f1"}}) {
    castVote {
      stake
      voteFor {
        memberId
        stake
        status
      }
      castBy
      electionRoundId
    }
    inBlock
  }
}
```

result:
```
{
  "data": {
    "voteCastEvents": [
      {
        "castVote": {
          "stake": "500000",
          "voteFor": {
            "memberId": "515",
            "stake": "150000",
            "status": "FAILED"
          },
          "castBy": "5HWKe5kQKPuKopnWkfwMdULGNsZimaxaLv7HuHjfy2yir3f1",
          "electionRoundId": "00000006"
        },
        "inBlock": 622702
      },

```

So the original problem described doesn't exist. Now issue of why the vote didn't show up in the past-votes need to be checked by pioneer team. I'll try to have a quick look at the the pioneer code base and check the query for that page is correct.

**Comment by mnaamani** [comment]
Following up, a query below:
```
query {
  castVotes(where: {castBy_eq: "5HWKe5kQKPuKopnWkfwMdULGNsZimaxaLv7HuHjfy2yir3f1"}){
    castBy, electionRound {isFinished, cycleId}, stake
  }
}
```

The interesting value to note is the `isFinished` value. Past votes will only show when this value is true. https://github.com/Joystream/pioneer/blob/1adcfe0cb7d4bcf2695eeb80537621557279367f/packages/ui/src/council/hooks/useMyPastVotes.ts#L19

And this value only becomes true when the council normal/idle period ends.

As we can see at time of writing the council for cycle id 9 is in its "normal" period:
```
{
  "data": {
    "castVotes": [
      {
        "castBy": "5HWKe5kQKPuKopnWkfwMdULGNsZimaxaLv7HuHjfy2yir3f1",
        "electionRoundId": "00000006",
        "electionRound": {
          "isFinished": true,
          "cycleId": 6
        },
        "stake": "500000"
      },
      {
        "castBy": "5HWKe5kQKPuKopnWkfwMdULGNsZimaxaLv7HuHjfy2yir3f1",
        "electionRoundId": "00000007",
        "electionRound": {
          "isFinished": true,
          "cycleId": 7
        },
        "stake": "1000000"
      },
      {
        "castBy": "5HWKe5kQKPuKopnWkfwMdULGNsZimaxaLv7HuHjfy2yir3f1",
        "electionRoundId": "00000009",
        "electionRound": {
          "isFinished": false,
          "cycleId": 9
        },
        "stake": "1000000"
      }
    ]
  }
}
```

So I suspect for the original case, cycle id 6, where the user voted for a failed candidate (status = FAILED)
```
"voteFor": {
            "memberId": "515",
            "stake": "150000",
            "status": "FAILED"
          },
```

The user was expecting, rightly so, to be able to release their vote stake (as the title of the issue suggests)

So in conclusion there is no issue with vote being not picked up by query node, but instead the query used in pioneer needs to be modified to allow displaying votes for failed candidates even when `isFinished` is false.

cc: @thesan

**Comment by mnaamani** [comment]
It would appear that comments in https://github.com/Joystream/joystream/issues/3573 suggest that `isFinished = true` is expected at end of an election and council is in normal period. I will review the council/election mappings in QN. Either the mappings are wrong, or the interpretation of the meaning of the field is wrong.
